test/NVQPP/integration/incorrect_argument_kernel_builder.cpp

```diff
@@ -0,0 +1,23 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2023 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+// RUN: nvq++ %s 2>&1 | FileCheck %s
```

Comment:
This tests should go in AST-error and driven in the same way. You can combine the tests, make sure the line number is reported correctly, etc.